server/sonar-server/src/main/java/org/sonar/server/es/textsearch/ComponentTextSearchFeatureRepertoire.java

```diff
@@ -53,35 +57,54 @@ public QueryBuilder getQuery(ComponentTextSearchQuery query) {
   },
   PREFIX(CHANGE_ORDER_OF_RESULTS) {
     @Override
-    public QueryBuilder getQuery(ComponentTextSearchQuery query) {
-      return prefixAndPartialQuery(query.getQueryText(), query.getFieldName(), SEARCH_PREFIX_ANALYZER)
+    public Stream<QueryBuilder> getQueries(ComponentTextSearchQuery query) {
+      List<String> tokens = split(query.getQueryText());
```

Comment:
The split is executed multiple times, once per search criteria. It should be part of `ComponentTextSearchQuery`.